src/services/graphql/resolvers/index.js

```diff
@@ -64,104 +48,65 @@ export default app => {
+      query: Object.assign({}, transformRegionArguments(regionArguments), {
+        $skip: 0,
+        $limit: 0
+      })
+    })
+
+    return {
+      ...args,
+      total: regions.total
+    }
+  }
+
+  const regionResultResolver = async (obj, args) => {
+    const regions = await app.service('regions').find({
+      query: Object.assign(
+        {},
+        transformRegionArguments(args),
+        transformPaginationArguments(obj)
+      )
+    })
+    return regions.data
   }
 
   return {
     Query: {
-      region: async (obj, args, context, info) => {
-        const valueAttributes = getFieldsFromInfo(info.fieldNodes[0])
-
-        const transformedRegionArguments = transformRegionArguments(args)
-
-        const transformedValueAttributes = transformValueAttributes(
-          valueAttributes
-        )
-
-        // regions
-        const region = await app.service('regions').get(args.id)
-
-        // statistics
-        context.data =
-          transformedValueAttributes.length > 0
-            ? await app.service('genesapiQuery').find({
-                index: elasticSearchIndex,
-                args: transformedRegionArguments,
-                fields: transformedValueAttributes
-              })
-            : []
-        context.regionArguments = transformedRegionArguments
-
-        return region
-      },
-      allRegions: async (obj, args, context, info) => {
-        const regionSelections = info.fieldNodes[0].selectionSet.selections.find(
-          f => f.name.value === 'regions'
-        )
-
-        const regionArguments = regionSelections.arguments.reduce(
-          (acc, curr) => {
-            acc[curr.name.value] = curr.value.value
-            return acc
-          },
-          {}
-        )
-
-        const valueAttributes = regionSelections
-          ? getFieldsFromInfo(regionSelections)
-          : []
-
-        const transformedValueAttributes = transformValueAttributes(
-          valueAttributes
-        )
-        const transformedRegionArguments = transformRegionArguments(
-          regionArguments
-        )
-        const transformedPaginationArguments = transformPaginationArguments(
-          args
-        )
-
-        // regions
-        const regions = await app.service('regions').find({
-          query: Object.assign(
-            {},
-            transformedRegionArguments,
-            transformedPaginationArguments
-          )
-        })
-
-        const region_id = regions.data.map(r => r.id)
-
-        // statistics
-        context.data =
-          valueAttributes.length > 0
-            ? await app.service('genesapiQuery').find({
-                index: elasticSearchIndex,
-                args: { ...transformedRegionArguments, region_id },
-                fields: transformedValueAttributes
-              })
-            : []
-
-        context.regionArguments = transformedRegionArguments
-
-        return {
-          page: regions.skip,
```

Comment:
Probably a minor thing: During manually testing, I noticed that in the results both `page` and `itemsPerPage` are `null` when they are not explicitly set in the attributes. Is this the expected behavior? I think it would make more sense if they would still show where we are in pagination:

```
      "itemsPerPage": 10,
      "page": 2,
```

Reply:
Good catch! Totally agree, added that.